### ISSUE TITLE
Unable to compile ec.rs: "error[E0425]: cannot find function `_mm_*` in this scope"

### ISSUE THREAD

**Issue by LigH-de** [xiph/rav1e#1179]
Trying to compile rav1e in media-autobuild_suite (MSYS2/MinGW) for both Win32 and Win64 targets, fails already in the Win32 pass.

Brief crop:
```
error[E0425]: cannot find function `_mm_and_si128` in this scope
   --> src\ec.rs:399:66
    |
399 |       let out_v = _mm_or_si128(_mm_andnot_si128(mask, shrunk_v), _mm_and_si128(mask, expanded_v));
    |                                                                  ^^^^^^^^^^^^^ not found in this scope
help: possible candidates are found in other modules, you can import them into scope
    |
13  | use core::arch::x86::_mm_and_si128;
    |
13  | use std::arch::x86::_mm_and_si128;
    |

error[E0425]: cannot find function `_mm_storel_epi64` in this scope
   --> src\ec.rs:400:7
    |
400 |       _mm_storel_epi64(cdf.as_mut_ptr() as *mut __m128i, out_v);
    |       ^^^^^^^^^^^^^^^^ not found in this scope
help: possible candidates are found in other modules, you can import them into scope
    |
13  | use core::arch::x86::_mm_storel_epi64;
    |
13  | use std::arch::x86::_mm_storel_epi64;
    |

error[E0412]: cannot find type `__m128i` in this scope
   --> src\ec.rs:400:49
    |
400 |       _mm_storel_epi64(cdf.as_mut_ptr() as *mut __m128i, out_v);
    |                                                 ^^^^^^^ not found in this scope
help: possible candidates are found in other modules, you can import them into scope
    |
13  | use core::arch::x86::__m128i;
    |
13  | use std::arch::x86::__m128i;
    |

error: aborting due to 18 previous errors

Some errors occurred: E0412, E0425.
```

Complete log files: [logs.zip](https://0x0.st/zP5l.zip)

**Comment by lu-zero** [comment]
Please update to the current stable rust.

**Comment by LigH-de** [comment]
In theory, MABS should have updated rust before trying to compile ... seems that just running rust_update is not sufficient to update rust.

**Comment by Arnavion** [comment]
@lu-zero OP is compiling for `cfg(target = "x86")`, but [the code](https://github.com/xiph/rav1e/blob/74edd38357ad52a3eb871b9a98ef5b6e398f84cc/src/ec.rs#L381-L382) only imports the intrinsics for `cfg(target = "x86_64")`. It probably needs a `#[cfg(target_arch = "x86")] use std::arch::x86::*;` or it needs to have both `cfg(target = "x86_64")` and `target_feature(enable = "sse2")` on the SIMD-using block, depending on what the code's intention is.

**Comment by lu-zero** [comment]
We should add to the CI a x86-windows target I guess =/

**Comment by tdaede** [comment]
Whoops, I had intended for that line to be x86_64 only. (technically it can work for x86 too, but it needs runtime selection)

**Comment by tdaede** [comment]
p.s. this doesn't need a x86-windows target, any x86 target will hit the issue.

**Comment by LigH-de** [comment]
Unfortunately, this fix is probably incomplete.

```
error[E0599]: no function or associated item named `update_cdf_4_sse2` found for type `ec::WriterBase<S>` in the current scope
   --> src\ec.rs:562:13
    |
101 | pub struct WriterBase<S> {
    | ------------------------ function or associated item `update_cdf_4_sse2` not found for this
...
562 |       Self::update_cdf_4_sse2(cdf, s);
    |       ------^^^^^^^^^^^^^^^^^
    |       |
    |       function or associated item not found in `ec::WriterBase<S>`
```

**Comment by LigH-de** [comment]
Fixed with bda810b66b57c0c128d5bc8f689b0db802f80c77

**Comment by lu-zero** [comment]
Great :)
